inst/qml/predictiveAnalytics.qml

```diff
@@ -371,51 +375,68 @@ Form
-										{label : qsTr("prophet - regression"), value: "prophetReg"},
-										//{label : qsTr("prophet - regression + lag"), value: "prophetRegLag"},
-										//{label : qsTr("xgboost - regression"), value: "xgboostReg"},
-										//{label : qsTr("xgboost - regression + lag"), value: "xgboostRegLag"},
-										{label : qsTr("bart - regression"), value: "bartReg" },
-										{label : qsTr("bart - regression + lag"), value: "bartRegLag"}
-										//{label : qsTr("bart - stack"), value: "bartStackReg"}
+					name:"availableModels"
+					// at the current moment, the user can choose models from a variety of predefined models
+					// some of these models contain a regressive component or lagged values
+					// but they only work if the user provided covariates or created some in the feature engineering section
+					// to streamline the user experience, this function dynamically changes which models are available in the qml menu
+					function getAvailableModels()
+					{
+						const models = []
 
+						if(!dependent.count > 0 && !time.count > 0)
+							return([{values: models}])
 
 
-										]
-									}]
-								}
-								AssignedVariablesList
-								{
-									name: "selectedModels"
-									id: selectedModels
+						// 'pure' time series models that only depend on time variable
+ 						// triggered if ready == T as time and dependent are needed for that
+						if(dependent.count > 0 && time.count > 0)
+						{
+							models[0] = {label : qsTr("linear regression - y ~ time"), value: "lmSpike"}
+							models[3] = {label : qsTr("bsts - linear trend model"), value: "bstsLinear"}
+							models[5] = {label : qsTr("bsts - autoregressive model"), value: "bstsAr"}
+							models[7] = {label : qsTr("prophet"), value: "prophet"}
+						}
+						// if covariates are provided or time based features are created
+						if(featEngAutoTimeBased.checked || factors.count > 0 || covariates.count > 0)
+						{
+							models[1] = {label : qsTr("linear regression - regression"), value: "lmSpikeReg"}
+							models[4] = {label : qsTr("bsts - linear trend model - regression"), value: "bstsLinearReg"}
+							models[6] = {label : qsTr("bsts - autoregressive model - regression"), value: "bstsArReg"}
+							models[8] = {label : qsTr("prophet - regression"), value: "prophetReg"}
+							models[9] = {label : qsTr("bart - regression"), value: "bartReg" }
 
-								}
+						}
+						//extra model for artificial lags because it can increase computation time a lot
+						if(featEngLags.value > 0)
+						{
+							models[2] = {label : qsTr("linear regression - regression + lag"), value: "lmSpikeRegLag"}
+							models[10] = {label : qsTr("bart - regression + lag"), value: "bartRegLag"}
+						}
+						// function returns one empty value and i haven't figured out why, so this is workaround
+						const modelsFiltered = models.filter((obj) => obj.label !== '')
 
+						const modelsList = [{values: modelsFiltered}]
+						return(modelsList)
+					}

```

Comment:
This function filters the available prediction models depending on whether they are possible based on the users input. For example one shouldn't be able to add a regression model like BART when no covariates are provided.
However, I discovered that the (selected) models in the qml elements are not saved when the JASP file is reopened. So one doesn't know anymore what models were selected for verification and prediction (although the JASP output/results persist).

I will try to fix this, please wait with approving this PR until then.

Reply:
I fixed it by creating a invisible availableVariableList that uses the function to make the right models available and then just use this as a source for a second availableVariableList that the user can see and use. this is a bit hacky, but works..